### PR TITLE
0.8 client decode jwt

### PR DESCRIPTION
**Pull request by marshallswain**
This adds a couple of util methods:
- `payloadIsValid` returns true if the payload hasn't expired.
- `decodeJWT` takes a token and returns the payload if it was valid.

`getJWT` has also been updated to only return the token if the token payload is valid.

Finally, `decodeJWT` is added to the `app` object, so you can do `app.decodeJWT(app.get('token'))` to get the token payload.  It's the same as the `decodeJWT` util, so it only returns a payload if it was valid.

Includes tests.  Will need docs before release:

### CODE REVIEW

**Comment by daffl** [comment]
Should we make it the same as on the server where we now have

```js
app.authentication.options // authOptions
app.authentication.verifyJWT // decode + verify
app.authentication.getJWT // get token
```

For the server `app.authentication` is an instance of this class: https://github.com/feathersjs/feathers-authentication/blob/0.8/src/base.js

**Comment by marshallswain** [comment]
Definitely.  I'll update it tomorrow.

**Comment by marshallswain** [comment]
@feathersjs/core-team do we want to keep the logout method where it is at `app.logout()`, or move it to `app.authentication.logout()`?

**Comment by ekryski** [comment]
hmm. I think keep it where it's at. We have `app.authenticate` as well.

**Comment by slajax** [comment]
I like having alias easily accesssible on app. That said, its more important to me that it is consistent if there is a client / server implementation.

**Comment by marshallswain** [comment]
Ok, so we'll have.
```
app.authenticate()
app.logout()
app.authentication.options // authOptions
app.authentication.verifyJWT // decode + verify
app.authentication.getJWT // get token
```

**Comment by slajax** [comment]
:+thumbsup

**Comment by ekryski** [comment]
Ya let's go with that. Once you get those changes in @marshallswain ping one of us for a quick ship.

**Comment by marshallswain** [comment]
Will do.  Also, I'm making both `verifyJWT` and `getJWT` sync, which won't match the server API (where both return promises), but I think promises will unnecessarily complicate things.  Let me know if you feel different.

**Comment by marshallswain** [comment]
@ekryski we're not returning the user in the authenticate requests anymore, right?  So this stuff should be able to come out: https://github.com/feathersjs/feathers-authentication/blob/0.8-client-decode-jwt/src/client/index.js#L54

**Comment by ekryski** [comment]
@marshallswain correcto. Will have to be a separate request by the client explicitly because you may not be authenticating a user. Could be a device, organization, toaster, whatever...

**Comment by marshallswain** [comment]
Other than what we discussed above, the only additional change is that verifyJWT now allows passing an object with a token property on it, which matches the server.  This should be ready to ship unless anybody finds any issues.  If not, please pull the trigger and release.  I'm retiring for the night. 🌙 😴

File: .travis.yml [hunk 1]
```diff
@@ -3,3 +3,5 @@ node_js:
   - 'node'
   - 'iojs'
   - '0.12'
+code_climate:
+  repo_token: 02ea69f85221efd3051a19e882c87cebc07522bf321374186bb6116301569e5e
```
Comment (ekryski) [inline .travis.yml]: This is slightly different to get it working but I will update.

**Comment by ekryski** [comment]
Nice! You pulled out the permissions stuff too. :shipit:

**Pull request merged**